src/main/java/Duke.java

```diff
@@ -1,10 +1,174 @@
+                str = str.substring(5, str.length());
+                Todo t = new Todo(str);
+                addToList(t, Tasks, j);
+                j++;
+                printTask(t, j);
+            }else if(str.contains("deadline")) {
+                if(!str.contains("/by")){
+                    printString("bad format");
+                }else {
+                    String time = str.substring(str.indexOf("/") + 3, str.length());
+                    if (str.length() - 9 + 11 > maxlength) maxlength = str.length() - 9 + 11;//- length of "deadline " + "1. [X][X]" + "(xx:" + ")"
+                    Deadline t = new Deadline(str.substring(9, str.indexOf("/") - 1), time);
+                    addToList(t, Tasks, j);
+                    j++;
+                    printTask(t, j);
+                }
+            }else if(str.contains("event")) {
+                if(!str.contains("/at")){
+                    printString("bad format");
+                }else {
+                    String time = str.substring(str.indexOf("/") + 3, str.length());
+                    if (str.length() - 6 + 11 > maxlength) maxlength = str.length() - 6 + 11;
+                    Event t = new Event(str.substring(6, str.indexOf("/") - 1), time);
+                    addToList(t, Tasks, j);
+                    j++;
+                    printTask(t, j);
+                }
+            }else if(str.contains("done")) {
+                String numberOnly = str.replaceAll("[^0-9]", "");
+                int num = Integer.parseInt(numberOnly);
+                if (num <= j) {
+                    Tasks[num - 1].setDone(true);
+                    printDone(Tasks[num - 1]);
+                } else {
+                    str = "not assigned yet";
+                    printString(str);
+                }
+            }else{
+                if(str.length() + 9 > maxlength) maxlength = str.length() + 9;
+                Task t = new Task(str);
+                addToList(t, Tasks, j);
+                j++;
+                printTask(t, j);
+            }
+        }
+    }
+
+    public static void addToList(Task t, Task[] list, int num){
+        list[num] = t;
+    }
+
+    public static void printString(String str){
+        int length = str.length();
+        System.out.print("      ");
+        for(int i = 0; i < length; i++) System.out.print("_");
+        System.out.println("");
+        System.out.println("     |" + str + "|");
+        System.out.print("     |");
+        for(int i = 0; i < length; i++) System.out.print("_");
+        System.out.println("|");
+    }
+
+    public static void printDone(Task t){
+        String str = "Nice! I've marked this task as done: ";
+        int length = Math.max(t.getLength(), str.length());
+        System.out.print("      ");
+        for(int i = 0; i < length; i++) System.out.print("_");
+        System.out.println("");
+        System.out.print("     |" + str);
```

Comment:
Considering that the indentation at the start of each string happens to always be "     |", could try defining that particular string as a constant instead such as TAB